classes/06class/exercises/c06-serverless01/alanlima/terraform/iam.tf

```diff
@@ -0,0 +1,122 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "lambda" {
+  name               = "${var.project}_lambda_role"
+  tags               = var.common_tags
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+                "Service": "lambda.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+        }
+    ]
+}
```

Comment:
For IAM permission statements, you can use the resource [`iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) to define your statements and get some validation for free in terraform.

Reply:
I saw this resource another day but didn't play with it yet. I will give it a go ;)